src/main/java/sg/dex/starfish/DataAsset.java

```diff
@@ -61,11 +67,88 @@ public default boolean isDataAsset() {
 
 		return buffer.toByteArray();
 	}
-	
+
 	/**
 	 * Gets the size of this data asset's content
+	 *
 	 * @return The size of the asset in bytes
 	 */
 	public abstract long getContentSize();
 
+
+	/**
+	 * This method is to validate the hash of the asset content .
+	 * it will calculate the hash of the content of an Asset
+	 * then compare with the hash value included in metadata,
+	 * if both are not the same , StarfishValidation Exception will be thrown
+	 * @return boolean  true if the content is valid else false
+	 *
+	 * @throws StarfishValidationException if hash content is not matched , exception will be thrown
+	 */
+	public default boolean validateContentHash() {
+
+		Object contentHashFromMetadata =  this.getMetadata().get(Constant.CONTENT_HASH);
+		if(null == contentHashFromMetadata){
+            throw new StarfishValidationException("Content hash is not included in the metadata");
+        }
+
+            String contentHash = Hex.toString(Hash.sha3_256(Utils.stringFromStream(this.getContentStream())));
+		if (null != contentHashFromMetadata && !contentHashFromMetadata.toString().equals(contentHash)) {
+			throw new StarfishValidationException("Failed to validate content hash");
+		}
+		return true;
+	}
+
+	/**
+	 * This method is used to calculate the hash of the content by using keccak256 hashing algorithm.
+	 *
+	 * @return the content of hash as string
+	 */
+
+	public default String getContentHash() {
+
+		return Hex.toString(Hash.sha3_256(this.getContent()));
+	}
+
+	/**
+	 * This method is to include the content of hash in the asset metadata.
+	 * Hash of the content will be calculated based on keccak256 hashing algo , and the hash content will
+	 * be included in the asset metadata.
+	 * This hash content will be used to validate the integrity of asset content
+	 *
+	 * @return respective data asset sub class.
+	 */
+	public default DataAsset includeContentHash() {
+
+		// check if the hash content is already present
+		if(null != this.getMetadata().get(Constant.CONTENT_HASH)){
+
+			if(validateContentHash()){
+				return this;
+			}
+		}
+
+		Map<String, Object> metaMap = this.getMetadata();
+		metaMap.put(Constant.CONTENT_HASH, getContentHash());
+
+		if (this instanceof DataAsset){
```

Comment:
when will this _not_ be an instance of `DataAsset`?  Again, throw an exception in the `else` part of this if-else section